src/main/java/seedu/address/logic/parser/visit/AddVisitCommandParser.java

```diff
@@ -45,7 +47,13 @@ public AddVisitCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddVisitCommand.MESSAGE_USAGE), pe);
         }
 
-        LocalDate date = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get());
+        LocalDate date;
+        try {
+            date = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get());
+        } catch (ParseException pe) {
+            throw new WrongDateFormatException(String.format(MESSAGE_INVALID_DATE_FORMAT,
+                    AddVisitCommand.MESSAGE_USAGE), pe);
+        }
```

Comment:
With the comments above, this can be implemented as a single line instead.

date = ParserUtil.parseDate(argMultimap.getValue(PREFIX_DATE).get());

Reply:
To be consistent with the code for parsing Index from line 42 to 47, I think it is better to still put it in a try catch block to catch an ParseException and throw new ParseException with a new message.

Reply:
Sure. Understood